gdnative-core/src/vector2.rs

```diff
@@ -15,6 +14,156 @@ impl FromVariant for Vector2 {
+    fn direction_to(self, point: Point2) -> Self;
+    /// Returns the distance to `point`.
+    fn distance_to(self, point: Point2) -> Length;
+    /// Returns the squared distance to `point`. Prefer this function over distance_to if you
+    /// need to sort vectors or need the squared distance for some formula.
+    fn distance_squared_to(self, point: Point2) -> Length;
+    /// Returns the vector projected onto the `other` vector.
+    fn project(self, other: Self) -> Self;
+    /// Returns the vector reflected from a plane defined by the given `normal`.
+    fn reflect(self, normal: Self) -> Self;
+    /// Returns the vector rotated by `angle` radians.
+    fn rotated(self, angle: Angle) -> Self;
+    /// Returns the component of the vector along a plane defined by the given normal.
+    fn slide(self, normal: Self) -> Self;
+    /// Returns the vector snapped to a grid with the given size.
+    fn snapped(self, by: Self) -> Self;
+    /// Returns a perpendicular vector.
+    fn tangent(self) -> Self;
+}
+
+impl Vector2Godot for Vector2 {
+    #[inline]
+    fn angle_to(self, other: Self) -> Angle {
+        Angle::radians(Trig::fast_atan2(self.cross(other), self.dot(other)))
+    }
+
+    #[inline]
+    fn angle_to_point(self, point: Point2) -> Angle {
+        Angle::radians(Trig::fast_atan2(self.y - point.y, self.x - point.x))
+    }
+
+    #[inline]
+    fn aspect(self) -> f32 {
+        self.x / self.y
+    }
+
+    #[inline]
+    fn bounce(self, normal: Self) -> Self {
+        let normal = normal.normalize();
+        normal * ((-2.0) * self.dot(normal)) + self
+    }
+
+    #[inline]
+    fn clamped(self, length: Length) -> Self {
+        if self.length() > length.get() {
+            self.normalize() * length.get()
+        } else {
+            self
+        }
+    }
+
+    #[inline]
+    fn cubic_interpolate(self, b: Self, pre_a: Self, post_b: Self, t: f32) -> Self {
+        let v0 = pre_a;
+        let v1 = self;
+        let v2 = b;
+        let v3 = post_b;
+
+        let t2 = t * t;
+        let t3 = t2 * t;
+
+        ((v1 * 2.0)
+            + (-v0 + v2) * t
+            + (v0 * 2.0 - v1 * 5.0 + v2 * 4.0 - v3) * t2
+            + (-v0 + v1 * 3.0 - v2 * 3.0 + v3) * t3)
+            * 0.5
+    }
+
+    #[inline]
+    fn direction_to(self, point: Point2) -> Self {
+        (point - self).to_vector().normalize()
+    }
+
+    #[inline]
+    fn distance_to(self, point: Point2) -> Length {
+        let squared = (self.x - point.x).powi(2) + (self.y - point.y).powi(2);
+        Length::new(squared.sqrt())
+    }
+
+    #[inline]
+    fn distance_squared_to(self, point: Point2) -> Length {
+        Length::new((self.x - point.x).powi(2) + (self.y - point.y).powi(2))
+    }
+
+    #[inline]
+    fn project(self, other: Self) -> Self {
+        let v1 = other;
+        let v2 = self;
+        v2 * (v1.dot(v2) / v2.dot(v2))
+    }
+
+    #[inline]
+    fn reflect(self, normal: Self) -> Self {
```

Comment:
I'm trying to understand this code. I see that it was copied from the C++ bindings, but it looks wrong to me.
[Godot's implementation](https://github.com/godotengine/godot/blob/bd7b2354c5384527ed73b5d13cee61d85f38fec6/core/math/vector2.cpp#L203) is different, and if I try it with simple values like `self = (1, 3)` and `normal = (0, -1)` I get `(6, 17)` which is not the expected result `(-1, 3)`.

` a.reflect(b) == -a.bounce(b)` so you could implement it that way. I like it even better because reflect confuses me (the name makes me think of the same behavior as bounce), so seeing in the code one defined in terms of the other helps understand the distinction.